### PR TITLE
Start of .gitignore

### PR DESCRIPTION
**Pull request by hanchenye**
Added the .gitignore file to the root. Not sure how to deal with the platforms directory though. Does it make sense to factor out a new repository to maintain the platforms?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/build*
+.vscode
+__pycache__
+
+/platforms
```
Comment (stephenneuendorffer) [inline .gitignore:5]: I think platforms are a submodule, so shouldn't be a problem?

**Review by stephenneuendorffer — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (hanchenye) [reply .gitignore:5]: The vck190_bare platform is not a submodule, which generates build files of petalinux and vivado after making.

Comment (stephenneuendorffer) [reply .gitignore:5]: It generates them in the platforms directory? That seems.. wrong...

Comment (hanchenye) [reply .gitignore:5]: Right. Based on the [document](https://xilinx.github.io/mlir-aie/Platform.html), we do `make all` under `platforms/vck190_bare` to built the petalinux and vivado project. The generated files are all under `platforms/vck190_bare`.

Comment (stephenneuendorffer) [reply .gitignore:5]: Can you make the ignore pattern more selective then?  Ultimately, this seems like something that should be fixed, though. so we're not generating things in the source tree.

**Review by stephenneuendorffer — APPROVED** [review]

**Pull request merged**